source/slang/slang-check-overload.cpp

```diff
@@ -883,7 +883,8 @@ namespace Slang
 
         for( auto constraintDecl : genericDeclRef.getDecl()->getMembersOfType<GenericTypeConstraintDecl>() )
         {
-            DeclRef<GenericTypeConstraintDecl> constraintDeclRef = m_astBuilder->getGenericAppDeclRef(genericDeclRef, substArgs, constraintDecl).as<GenericTypeConstraintDecl>();
+            DeclRef<GenericTypeConstraintDecl> constraintDeclRef = m_astBuilder->getGenericAppDeclRef(
+                genericDeclRef, newArgs.getArrayView(), constraintDecl).as<GenericTypeConstraintDecl>();
```

Comment:
Is this change to fix a bug?
When I compare to the old code, `substArgs` passed in won't change, while `newArgs` keep changing during iteration.

Reply:
Not technically a bug, but this is to allow subtype witness finding to make use of the already solved args as we make forward progress. This only matters when we have dependent generic parameters and wasn't an issue before.

Reply:
Nice to know.